### PR TITLE
[P3] [Localization] Added Snow Worker Keys and Dialogue for En, Es, It, Fr, De, Ko, / Suppressed Female

### PR DESCRIPTION
**Pull request by Snailman11**
## What are the changes the user will see?

-Snow Worker will have translated dialogue in English, French, Spanish, Italian, German, and Korean.
-Brazilian Portuguese, ZH_CN, ZH_TW, and Japanese, have placeholder English dialogue.

-Removed the option for Snow Worker to spawn alone as female. It was using the female worker sprite as there actually is no female snow worker.

-In double battles, the second snow worker will always spawn with a female name, despite using the default (male) sprite. This is likely because double battles force Male/Female pairs.

## Why am I making these changes?

Closes https://github.com/pagefaultgames/pokerogue/issues/4081

Link to Discord's Bug Report (Snow worker has no dialogue)
https://discord.com/channels/1125469663833370665/1281452532455047168

## What are the changes from a developer perspective?

-Added Dialogue Keys in **Dialogue.ts** to summon Single and Double Snow Worker "encounter" and "victory" lines.
Doubles Dialogue is not currently displayed but added doubles dialogues keys and lines if the doubles keys ever get called in the future.
**Did not add** "defeat" keys
-Added Translated Single and Doubles Snow worker "encounter", "victory", and "defeat" lines for English, French, Spanish, German, Italian, and Korean to their respective JSON files
-Added placeholder English Text for Japanese, ZH_CN, ZH_TW, PT_BR to their respective JSON files

-setHasGenders("Worker Female") has been **removed** from trainer-config.ts .
-Added "snow_worker.json" and "snow_worker.png", copied from "snow_worker_m.json" and "snow_worker_m.png"
-snow_worker_m.png, snow_worker_m.json, snow_worker_f.png and snow_worker_f.json, **still exist.**

## How to test the changes?
**Overrides.ts**
```
const overrides = {
STARTING_LEVEL_OVERRIDE: 8,
OPP_SPECIES_OVERRIDE: Species.PIDGEY,
STARTING_WAVE_OVERRIDE: 2,
OPP_LEVEL_OVERRIDE: 1
} satisfies Partial<InstanceType<typeof DefaultOverrides>>;
```

**Battle.ts**

Singles Dialogue
```
[2]: new FixedBattleConfig().setBattleType(BattleType.TRAINER)
    .setGetTrainerFunc(scene => new Trainer(scene, TrainerType.SNOW_WORKER, TrainerVariant.DEFAULT)),
```
Doubles Dialogue
```
[2]: new FixedBattleConfig().setBattleType(BattleType.TRAINER)
    .setGetTrainerFunc(scene => new Trainer(scene, TrainerType.SNOW_WORKER, TrainerVariant.DOUBLE)),
```
### Screenshots/Videos
<details>
  <summary>Expand for videos</summary>

English

https://github.com/user-attachments/assets/3aa047c7-06d6-426e-8907-86f3fce88a42

https://github.com/user-attachments/assets/e347f6a3-802f-4a1f-be23-ed0f9ed8f6ed

Spanish

https://github.com/user-attachments/assets/ff4d7b68-00d6-49f5-a549-37d10c062862

https://github.com/user-attachments/assets/8982f6dc-17a3-4fa0-bdd4-0fc3f527e151

French

https://github.com/user-attachments/assets/56aed76b-5ce9-4d75-8e04-67b7ee7b6138

https://github.com/user-attachments/assets/120ed604-ab31-48df-b096-4cd3b5dd5639

Italian

https://github.com/user-attachments/assets/e5dfc0d4-bcbb-4ea3-90da-7ae3e09afe7d

https://github.com/user-attachments/assets/4210d59b-fb20-4c03-b8ae-a98e2a04d152

German

https://github.com/user-attachments/assets/f99422b1-df2a-471e-bea9-116232f5fe2a

https://github.com/user-attachments/assets/621cdd1d-476f-4fc6-bf44-0a30f7b45d96

Korean

https://github.com/user-attachments/assets/2cfd6fbf-b6bd-42a8-b76e-6c1557bfd82a

https://github.com/user-attachments/assets/336593eb-26fa-416b-aca7-8e41531bc821

Japanese

https://github.com/user-attachments/assets/713183ad-95e8-49b2-a6a1-5cc73ddd9d92

https://github.com/user-attachments/assets/be38817c-5547-48b6-bf0e-1535662f0a85

Chinese (China)

https://github.com/user-attachments/assets/93a0046e-7fce-4fd0-96bc-3444a7a4691d

https://github.com/user-attachments/assets/c8348a1d-1760-47a3-90a7-bf761f8d6d72

Chinese (Taiwan)

https://github.com/user-attachments/assets/47c06826-db17-413a-8855-1b6c3ad0575e

https://github.com/user-attachments/assets/f0672cb1-8816-41c7-9cbf-f8429f39b3c5

Brazilian Portuguese

https://github.com/user-attachments/assets/ff826bd4-6d66-4d4a-9417-32bd5a34ad24

https://github.com/user-attachments/assets/410b5027-40fd-4f3a-8967-86730301c224

</details>

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/de/dialogue.json [hunk 1]
```diff
@@ -299,6 +299,28 @@
...
   },
+  "snow_worker": {
+    "encounter": {
+      "1": "Wenn es kalt wird, zieh dich warm an!\n$Das sagt uns der gesunde Menschenverstand."
+    },
+    "victory": {
+      "1": "Sich bis zum Gehtnichtmehr abzurackern, ist wohl auch eine Eigenart, die uns der gesunde Menschenverstand diktiert."
+    },
+    "defeat": {
+	    "1": "Lass mich dir einen Ratschlag geben. Ist ein Pokémon gefroren, kannst du es wieder auftauen, indem du eine Attacke vom Typ Feuer auf es anwendest!"
```
Comment (CodeTappert) [inline src/locales/de/dialogue.json:310]: Did you test this ingame? Seems very long and needs a $ i guess

**Comment by Adri1** [comment]
"-Removed the option for Snow Worker to spawn alone as female. It was using the female worker sprite as there actually is no female snow worker."
And why shoud they not exists in Rogue just because they dont in OG games? Personally I'm against removing them because of this reason.
+ You deleted females, bu you kept the female names.
image
image
image

**Review by CodeTappert — APPROVED** [review]

**Comment by Snailman11** [comment]
Here's what it looks like with \n$ appended to German and Spanish

https://github.com/user-attachments/assets/746d7d65-7eb9-45ba-8396-0bbd7ef32826


https://github.com/user-attachments/assets/c5db426a-1583-4230-8e4a-bfc49602a073

**Comment by Snailman11** [comment]
Here's how the Korean double text would display with ellipses by @returntoice 's suggestion

Doubles Encounter and Victory (changed . . . to ellipsis)

https://github.com/user-attachments/assets/3d01db38-cb3e-4986-a389-e786798fb402

Doubles Defeat (changed ... to ellipsis)

https://github.com/user-attachments/assets/06238dab-fe22-43e2-bf5d-a095e43c172f

Singles Defeat (added period)

https://github.com/user-attachments/assets/452a2109-8f30-4eca-8ea0-7b15f46c883d

File: src/locales/ja/dialogue.json [hunk 2]
```diff
@@ -316,6 +316,28 @@
...
+    "encounter": {
+      "1": "...\n$...R-ready to b-battle?"
+    },
+    "victory": {
+      "1": "...\n$...I'm sh-shivering!"
+    },
+    "defeat": {
+      "1": "...\n$...Kn-know what?\n...Th-this place is cold."
+    } 
+  },
```
Comment (flx-sta) [inline src/locales/ja/dialogue.json:340]: No English placeholders anymore
```suggestion
```

File: src/locales/pt_BR/dialogue.json [hunk 3]
```diff
@@ -311,6 +311,28 @@
...
+    "encounter": {
+      "1": "...\n$...R-ready to b-battle?"
+    },
+    "victory": {
+      "1": "...\n$...I'm sh-shivering!"
+    },
+    "defeat": {
+      "1": "...\n$...Kn-know what?\n...Th-this place is cold."
+    } 
+  },
```
Comment (flx-sta) [inline src/locales/pt_BR/dialogue.json:335]: No English placeholders anymore
```suggestion
```

File: src/locales/zh_CN/dialogue.json [hunk 4]
```diff
@@ -299,6 +299,28 @@
...
+    "encounter": {
+      "1": "...\n$...R-ready to b-battle?"
+    },
+    "victory": {
+      "1": "...\n$...I'm sh-shivering!"
+    },
+    "defeat": {
+      "1": "...\n$...Kn-know what?\n...Th-this place is cold."
+    } 
+  },
```
Comment (flx-sta) [inline src/locales/zh_CN/dialogue.json:323]: No English placeholders anymore
```suggestion
```

File: src/locales/zh_TW/dialogue.json [hunk 5]
```diff
@@ -299,6 +299,28 @@
...
+    "encounter": {
+      "1": "...\n$...R-ready to b-battle?"
+    },
+    "victory": {
+      "1": "...\n$...I'm sh-shivering!"
+    },
+    "defeat": {
+      "1": "...\n$...Kn-know what?\n...Th-this place is cold."
+    } 
+  },
```
Comment (flx-sta) [inline src/locales/zh_TW/dialogue.json:323]: No English placeholders anymore
```suggestion
```

**Review by flx-sta — CHANGES_REQUESTED** [review]

**Comment by flx-sta** [comment]
https://discord.com/channels/1125469663833370665/1281859097481711648/1288251320826068992

**Review by SangaraSorama — APPROVED** [review]

**Review by CodeTappert — APPROVED** [review]

**Pull request merged**